test/unit/sfErrorTest.ts

```diff
@@ -70,6 +133,55 @@ describe('SfError', () => {
       expect(mySfError).to.be.an.instanceOf(SfError);
       expect(mySfError).to.equal(existingSfError);
     });
+
+    describe('handling "other" stuff that is not Error', () => {
+      it('undefined', () => {
+        const wrapMe = undefined;
+        const mySfError = SfError.wrap(wrapMe);
+        expect(mySfError).to.be.an.instanceOf(SfError);
+        expect(mySfError.message === 'An unexpected error occurred');
+        expect(mySfError.name === 'TypeError');
```

Comment:
Does it _really_ make sense than an `SfError.wrap(undefined).name==='TypeError'`?

Reply:
I think so.  You're passing something that is of an unexpected type and we can't do much with it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError
> A TypeError may be thrown when:
> an operand or argument passed to a function is incompatible with the type expected by that operator or function; or

What would you prefer?